### PR TITLE
Issue error when tagging templates and decorating with only-constructable entities

### LINKED ISSUE #10454: No error when templates are tagged by anything with only construct signatures
**TypeScript Version:**  1.8.36.0

Want to use function htmlEscape(literals, ...placeholders) {...}

but type  HTMLElement`${xyz}` (thanks to intellisense)

This should be reported as an error since the tag function is not valid. I often run into this because intellisense doesn't know better and offers up the HTMLElement isntead.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #10454.

@yuit @rbuckton can you review?

### CODE REVIEW

**Comment by yuit** [comment]
I don't get why it has to have both call signature and constructor signatures. Would having just call signatures not enough?

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11943,18 +11943,12 @@ namespace ts {
...
             // that the user will not add any.
             const callSignatures = getSignaturesOfType(apparentType, SignatureKind.Call);
-
             const constructSignatures = getSignaturesOfType(apparentType, SignatureKind.Construct);
-            // TS 1.0 spec: 4.12
-            // If FuncExpr is of type Any, or of an object type that has no call or construct signatures
-            // but is a subtype of the Function interface, the call is an untyped function call. In an
-            // untyped function call no TypeArgs are permitted, Args can be any argument list, no contextual
+
+            // TS 1.0 Spec: 4.12
```
Comment (yuit) [inline src/compiler/checker.ts]: Also you may want to update spec section

**Comment by DanielRosenwasser** [comment]
@yuit I guess the idea was always that if you have any call/construct signatures, you probably should have an appropriate signature for the call. Otherwise, if all you know is that something's a `Function`, `any` is the best you can do.

**Comment by mhegazy** [comment]
👍

**Pull request merged**